### PR TITLE
[[ Debug ]] Docs and tweaks for android debugging

### PR DESCRIPTION
**Pull request by montegoulding**
This patch documents how to debug and profile an android debug build using
Android Studio. It also makes some minor tweaks to our builds to take a
couple of steps that would need to be done each build out of the process.

### CODE REVIEW

File: util/set_android_debuggable.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -e
+
+INPUT="$1"
+
+if [ "$BUILDTYPE" == "Debug" ] ; then
+	sed -i.bak 's/android:debuggable="false"/android:debuggable="true"/g' "${INPUT}"
```
Comment (livecodefraser) [inline util/set_android_debuggable.sh]: I'm not 100% sure that this sed command will work with GNU sed (i.e on Linux). Unfortunately, the Vulcan build won't test it as it uses $BUILDTYPE==Fast.

**Review by livecodefraser — APPROVED** [review]
@livecode-vulcan review ok d73ea5e

@montegoulding could you make sure that building a debug Android engine on Linux works? Other than that, it looks good to me.

File: util/set_android_debuggable.sh (hunk 1 above)
Comment (montegoulding) [reply util/set_android_debuggable.sh]: I _think_ the sed command is ok but it didn't seem to do the actions in order so I'm combing the copy and set debuggable actions into the one script

**Comment by montegoulding** [comment]
Now tweaked to ensure it builds on linux

**Comment by runrevmark** [comment]
@montegoulding Can you rebase this so we can merge :)

**Comment by runrevmark** [comment]
(and maybe repoint at develop!)

**Comment by runrevmark** [comment]
haha - never mind - the PR base just needed updated :)

File: util/set_android_debuggable.sh (hunk 1 above)
Comment (runrevmark) [reply util/set_android_debuggable.sh]: @montegoulding I take it this was actioned? (Its not clear what change you suggested)

Comment (montegoulding) [reply util/set_android_debuggable.sh]: @runrevmark I think so but I'll double-check.

Comment (montegoulding) [reply util/set_android_debuggable.sh]: aha... still have it checked out on my linux VM and I managed to clobber my tweak when I rebased the branch on mac so will sort that in a sec

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 5fa5ed5

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 5fa5ed5fa4703029cdea0ea39ef3d1530d93569f

**Comment by livecode-vulcan** [comment]
😎 test success 5fa5ed5fa4703029cdea0ea39ef3d1530d93569f

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/746)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/155)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/155)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/155)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/155)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2272)
* try-community-universal-ios-iphoneos13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos13.4/builds/6)
* try-community-universal-ios-iphonesimulator13.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator13.4/builds/6)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2301)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2288)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2287)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2322)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2262)

**Pull request merged**